src/main/java/com/userservice/Service/UserService.java

```diff
@@ -0,0 +1,32 @@
+package com.userservice.Service;
+
+import com.userservice.Entity.Users;
+import com.userservice.Repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+@Service
+public class UserService {
+    @Autowired
+    private UserRepository userRepository;
+    public ResponseEntity<?> create(Users users){
+        Users usersSaved = userRepository.save(users);
+        return ResponseEntity.ok(usersSaved);
+    }
+    public ResponseEntity<?> getUserDetails(String username) {
+        Users users = userRepository.findByUsername(username);
```

Comment:
Please use the phone number with the country code to get the user details and also create a similar function in which we will be using ID to get the user details. As our application follows microservice architecture, they will be communicating with each other, so probably they will be using ID. That's why getting user details using phone number and ID are both required